src/keybaseca/bot/bot.go

```diff
@@ -144,3 +149,35 @@ func isConfiguredTeam(conf config.Config, teamName string, channelName string) b
 	}
 	return false
 }
+
+func buildAnnouncement(template, username, currentTeam string, teams []string) string {
+	replacements := map[string]string{
+		"{USERNAME}":     username,
+		"{CURRENT_TEAM}": currentTeam,
+		"{TEAMS}":        strings.Join(teams, ", "),
+	}
+
+	templatedMessage := template
+	for templateStr, templateVal := range replacements {
+		templatedMessage = strings.Replace(templatedMessage, templateStr, templateVal, -1)
+	}
+
+	return templatedMessage
+}
+
+func sendAnnouncementMessage(conf config.Config, kbc *kbchat.API) error {
+	if conf.GetAnnouncement() == "" {
+		// No announcement to send
+		return nil
+	}
+	for _, team := range conf.GetTeams() {
+		announcement := buildAnnouncement(conf.GetAnnouncement(), kbc.GetUsername(), team, conf.GetTeams())
```

Comment:
if you can imagine adding more things to template, i'd suggest turning this into an object that wraps these three things and then passing that. not a big deal or blocker for now, just the kind of thing that makes extending in obvious ways not require changing method signatures.

Reply:
Updated. I don't necessarily see any other things we'd want to template, but still seemed worth doing.